src/provider/provider_file_memory.c

```diff
@@ -454,7 +454,7 @@ static umf_result_t file_alloc_aligned(file_memory_provider_t *file_provider,
     size_t new_offset_fd =
         file_provider->offset_fd + new_offset_mmap - file_provider->offset_mmap;
 
-    if (file_provider->size_mmap - new_offset_mmap < size) {
+    if (file_provider->size_mmap < size + new_offset_mmap) {
```

Comment:
so there was an underflow here? I see similar case at line 432 - should it also be fixed?

Reply:
1) Yes, there was an underflow here.
2) No, there should be always: `file_provider->offset_mmap <= file_provider->size_mmap` - I added an assert for that.